src/Context/ForkContext.php

```diff
@@ -0,0 +1,170 @@
+<?php declare(strict_types=1);
+
+namespace Amp\Parallel\Context;
+
+use Amp\ByteStream\StreamChannel;
+use Amp\Cancellation;
+use Amp\Parallel\Context\Internal\AbstractContext;
+use Amp\Parallel\Ipc\IpcHub;
+use Amp\Serialization\NativeSerializer;
+use Amp\Serialization\Serializer;
+use Amp\TimeoutCancellation;
+use Revolt\EventLoop;
+use Revolt\EventLoop\Driver\UvDriver;
+
+/**
+ * USE AT YOUR OWN RISK! This context is not used by default in {@see DefaultContextFactory} because the timing of its
+ * creation must be purposeful and situational.
+ *
+ * Forking is not recommended at arbitrary points in an application since the entire state of the parent process is
+ * inherited into the child process, including the event-loop!
+ *
+ * Forking very early in an application, may be beneficial to copy state from the parent that was expensive to set up.
+ *
+ * This context is NOT compatible with event-loop extensions such as ext-uv or ext-ev.
+ *
+ * @template-covariant TResult
+ * @template-covariant TReceive
+ * @template TSend
+ * @extends AbstractContext<TResult, TReceive, TSend>
+ */
+final class ForkContext extends AbstractContext
+{
+    private const DEFAULT_START_TIMEOUT = 5;
+
+    public static function isSupported(): bool
+    {
+        return \function_exists('pcntl_fork')
+            && !EventLoop::getDriver() instanceof UvDriver;
```

Comment:
Docs say it's not compatible with extension based event loops, but we only check for UV here.

Reply:
Yeah, this needs fixing, @trowski mind bumping